### ISSUE TITLE
LibreElec 9.2 Raspberry Pi 4 - max. Resolution 960x520 | 960x400

### ISSUE THREAD

**Issue by Philps** [Sandmann79/xbmc#413]
**Addon used**
- [x ] Amazon VOD (plugin.video.amazon-test)
- Addon Version [0.8.1]

**Prime Video Account (select one)**
- [x ] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware: Raspberry Pi 4
- OS Version: LibreElec 9.2
- Kodi Version 18.5

**Upload Logs**
https://pastebin.com/yWdPrFnJ

**Describe the bug**
Input Stream Adaptive Settings:
Min Bandwith. 3 500 000
Max Bandwith 0
Max Resolution Decoder 1080p
Max Secure Decoder 1080p
Stream Selection Auto & Manual tested
Override HDCP Status Yes & No tested

I am not sure if this is a limitation of LibreElec ( Debian) in combination with Widevine,DRM,  a Input Stream Adaptive Bug or a Amazon VOD Bug.

From the logs there is no failure. 11 Streams (Low Res up to 960x5xx) being listed but no HD and higher Res. 

I am fairly new to Pi and used Android on FireTv Box before which worked like charm. 

Only found this bug https://github.com/Sandmann79/xbmc/issues/205 but doesn't seem to be related and years old.

**Comment by Philps** [comment]
Could this be the reason https://github.com/Sandmann79/xbmc/pull/295 ?

**Comment by Varstahl** [comment]
Since last year google requires VMP for anything HD and up (720p+). Since Kodi and the majority of open software can't get neither VMP certification nor a reply when they apply for it, there's no way to play it natively. On Android/FireStick you can use Kodi and then playback on the native app to get up to 4K, but on plain Kodi that's just not possible.

TL;DR: if the hardware doesn't have a specific hardware chip, no 4k, if the software doesn't have VMP certification no HD+.

**Comment by fatshotty** [comment]
Hi there,
I understand the reason about VMP, but I have a question (curiosity): why other addon such as "netflix" can stream in FHD (1080p) ? Has Netflix no check to VMP?

Thanks in advance and I'm sorry if my question is annoing you
Fat

**Comment by Varstahl** [comment]
It really depends on how they decide to provide the encryption. Widevine can be varied, and while Amazon and Google decided to be ultra-strict, possibly to sell their physical goods, Netflix didn't (possibly because they don't sell sticks, so they're not as interested).

It's just politics, there's no good reason why it shouldn't work. They just decided it to be that way. It wasn't like this in the past, they made it worse intentionally.

Also, this addon *can* playback up to 4k videos if I'm not mistaken, on hardware such as the Nvidia Shield, it just can't do it everywhere due to hardware/software restrictions.

**Comment by fatshotty** [comment]
thanks a lot for your explanation and thanks for good job with this addon ;)
